cmds/tail/tail.go

```diff
@@ -0,0 +1,259 @@
+// Copyright 2012-2017 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Tail prints the lasts 10 lines of a file. Can additionally follow the
+// the end of the file as it grows.
+//
+// Synopsis:
+//     tail [-f] [-n lines_to_show] [FILE]
+//
+// Description:
+//     If no files are specified, read from stdin.
+//
+// Options:
+//     -f: follow the end of the file as it grows
+//     -n: specify the number of lines to show (default: 10)
+
+// Missing features:
+// - follow-mode (i.e. tail -f)
+
+package main
+
+import (
+	"bytes"
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"syscall"
+)
+
+var (
+	flagFollow   = flag.Bool("f", false, "follow the end of the file")
+	flagNumLines = flag.Int("n", 10, "specify the number of lines to show")
+)
+
+// TailConfig is a configuration object for the Tail function
+type TailConfig struct {
+	follow   bool // enable follow-mode (-f)
+	numLines int  // specifies the number of lines to print (-n)
```

Comment:
why isn't this a uint?

Reply:
I did it to simplify the code for all the comparisons and other typed operations, and because there is a check for non-negativeness before passing it (even if I don't feel comfortable with checking values in outer functions that are needed in inner functions). I can go back to `uint` here though